handlers/auth.go

```diff
@@ -97,18 +98,27 @@ func AuthStateHandler(w http.ResponseWriter, r *http.Request) {
 
 	// SUCCESS!! they are authorized
 
+	// get the originally requested URL so we can send them on their way
+	requestedURL := session.Values["requestedURL"].(string)
+
 	// issue the jwt
+	var tokenstring string
+	if requestedURL == "" {
+		tokenstring, err = jwtmanager.NewVPJWT(user, customClaims, ptokens)
+	} else {
+		u, _ := url.Parse(requestedURL)
+		aud := domains.Matches(u.Host)
```

Comment:
1. Return 400 when `requestedURL` cannot be parsed
2. Return 403 when the returned `aud` is an empty string (i.e. the requested host is somehow not whitelisted)